foursight_core/react/api/aws_stacks.py

```diff
@@ -0,0 +1,167 @@
+    if stack:
+        result = _create_aws_stack_info(stack)
+        outputs = aws_get_stack_outputs(stack)
+        parameters = aws_get_stack_parameters(stack)
+        resources = aws_get_stack_resources(stack)
+        result["outputs"] = outputs
+        result["parameters"] = parameters
+        result["resources"] = resources
+    return result
+
+
+@memoize
+def aws_get_stack_outputs(stack_name_or_object: Union[str, object]) -> dict:
+    """
+    Returns the name/value outputs for the given AWS CloudFormation stack name.
+    Output values are obfuscated if the output name represents a sensitive value.
+    """
+    def output_id(element):
+        return element["OutputKey"]
+    result = {}
+    stack = _get_aws_stack_object(stack_name_or_object)
+    if stack and stack.outputs:
+        for stack_output in sorted(stack.outputs, key=output_id):
+            result[stack_output.get("OutputKey")] = stack_output.get("OutputValue")
+    return _obfuscate(result)
+
+
+@memoize
+def aws_get_stack_parameters(stack_name_or_object: Union[str, object]) -> dict:
+    """
+    Returns a name/value dictionary of the parameters for the given AWS CloudFormation stack name.
+    Parameter values are obfuscated if the parameter name represents a sensitive value.
+    """
+    def parameter_id(element):
+        return element["ParameterKey"]
+    result = {}
+    stack = _get_aws_stack_object(stack_name_or_object)
+    if stack and stack.parameters:
+        for stack_parameter in sorted(stack.parameters, key=parameter_id):
+            result[stack_parameter.get("ParameterKey")] = stack_parameter.get("ParameterValue")
+    return _obfuscate(result)
+
+
+@memoize
+def aws_get_stack_resources(stack_name_or_object: Union[str, object]) -> dict:
+    """
+    Returns a name/value dictionary of the resources for the given AWS CloudFormation stack name.
+    """
+    def resource_id(element):
+        return element.logical_resource_id
+    result = {}
+    stack = _get_aws_stack_object(stack_name_or_object)
+    if stack:
+        for stack_resource in sorted(list(stack.resource_summaries.all()), key=resource_id):
+            result[stack_resource.logical_resource_id] = stack_resource.resource_type
+    return _obfuscate(result)
+
+
+def _get_aws_stack_object(stack_name_or_object: Union[str, object]) -> Optional[object]:
+    """
+    Returns the boto3 Cloudformation stack object for the given stack name, if the given argument
+    is a string, or if the given argument is already a boto3 stack object, then return that value.
+    :param stack_name_or_object: String representing Cloudformation stack name or boto3 stack object.
+    :returns: The boto3 Cloudformation stack object or None if not found.
+    """
+    if type(stack_name_or_object).__name__ == "cloudformation.Stack":
+        return stack_name_or_object
+    elif not isinstance(stack_name_or_object, str):
+        return None
+    c4 = boto3.resource("cloudformation")
```

Comment:
huh. Are you thinking the "4" in "c4" is "for", like "cloud formation'? It could be used that way, I guess. For what it's worth, I think c4 stands for "cgap/4dn", so that makes this assignment look odd. I'd call the variable `cf` or even `cloudformation`, but I'm sure it's fine either way. I mostly just wanted to register the realization about what you were doing.